src/main/java/org/opentripplanner/middleware/trip_monitor/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,176 @@
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.trip_monitor.jobs.TripMonitorNotification;
+import org.opentripplanner.middleware.utils.DateUtils;
+import org.opentripplanner.middleware.utils.NotificationUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CheckMonitoredTrip implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(CheckMonitoredTrip.class);
+    private final MonitoredTrip trip;
+    private OtpDispatcherResponse injectedOtpResponse;
+    private final List<TripMonitorNotification> notifications = new ArrayList<>();
+
+    public CheckMonitoredTrip(MonitoredTrip trip) {
+        this.trip = trip;
+    }
+
+    public CheckMonitoredTrip(MonitoredTrip trip, OtpDispatcherResponse injectedOtpResponse) {
+        this.trip = trip;
+        this.injectedOtpResponse = injectedOtpResponse;
+    }
+
+    @Override
+    public void run() {
+        // Check if the trip check should be skipped (based on time, day of week, etc.)
+        if (shouldSkipMonitoredTripCheck(trip)) {
+            LOG.debug("Skipping check for trip: {}", trip.id);
+            return;
+        }
+        // Make a request to OTP with the monitored trip params or if this is in a test environment use the injected OTP
+        // response.
+        OtpDispatcherResponse otpDispatcherResponse = injectedOtpResponse != null
+            ? injectedOtpResponse
+            : OtpDispatcher.sendOtpPlanRequest(trip.queryParams);
+
+        if (otpDispatcherResponse.statusCode >= 400) {
+            // TODO: report bugsnag
+            LOG.error("Could not reach OTP server. status={}", otpDispatcherResponse.statusCode);
+            return;
+        }
+        Response otpResponse = otpDispatcherResponse.getResponse();
+        // TODO: Should null tripRequestId be fixed?
+        // TripSummary tripSummary = new TripSummary(otpDispatcherResponse.response.plan, otpDispatcherResponse.response.error, null);
+        // TODO: Find the specific itinerary to compare against. For now, just choose the first itin.
+        Itinerary itinerary = otpResponse.plan.itineraries.get(0);
+        // BEGIN CHECKS
+        // Check for notifications related to service alerts.
+        checkTripForNewAlerts(trip, itinerary);
+        // Check for notifications related to delays
+        checkTripForDelays(trip, itinerary);
+        // Check for notifications related to itinerary changes
+        checkTripForItineraryChanges(trip, itinerary);
+        // Add latest OTP response.
+        trip.addResponse(otpResponse);
+        Persistence.monitoredTrips.replace(trip.id, trip);
+        // Send notification to user.
+        sendNotification();
+    }
+
+    private void checkTripForItineraryChanges(MonitoredTrip trip, Itinerary itinerary) {
```

Comment:
Can the next 4 methods be in the order that they are called in the `run` method?